### PR TITLE
Add GetAllPartitionNames() to the SourceDB interface.

### PR DESCRIPTION
**Pull request by amit-yb**

### CODE REVIEW

File: yb_migrate/src/srcdb/mysql.go [hunk 1]
```diff
@@ -72,6 +72,10 @@ func (ms *MySQL) GetAllTableNames() []string {
 	return tableNames
 }
 
+func (ms *MySQL) GetAllPartitionNames(tableName string) []string {
+	panic("Not Implemented")
```
Comment (sanyamsinghal) [inline yb_migrate/src/srcdb/mysql.go:76]: @amit-yb is it okay to panic() as it shows the complete stack trace? or we should use utils.ErrExit() here

**Review by sanyamsinghal — APPROVED** [review]
LGTM, please respond to the comment before merging.

File: yb_migrate/src/srcdb/mysql.go (hunk 1 above)
Comment (amit-yb) [reply yb_migrate/src/srcdb/mysql.go:76]: In its current form this function should never be called. If it gets called for PG or MySQL, we will get this panic message right away during testing. So, users will never see this panic.

Comment (sanyamsinghal) [reply yb_migrate/src/srcdb/mysql.go:76]: got it, thanks!

**Pull request merged**